client/src/app/core/services/csv-export.service.ts

```diff
@@ -0,0 +1,118 @@
+import { BaseViewModel } from '../../site/base/base-view-model';
+import { Injectable } from '@angular/core';
+import { FileExportService } from './file-export.service';
+
+/**
+ * Handles all incoming and outgoing notify messages via {@link WebsocketService}.
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class CsvExportService {
+    protected exporter = new FileExportService();
+
+    /**
+     * empty constructor TODO
+     */
+    public constructor() {}
+
+    /**
+     * Saves an array of model data to a CSV.
+     * @param data Array of Model instances to be saved
+     * @param columns Column definitions (model property; optional csv title)
+     * @param filename name of the resulting file
+     * @param options
+     */
+    public saveAsCsv(
+        data: BaseViewModel[],
+        columns: { property: string; title?: string }[],
+        filename: string,
+        { lineSeparator = '\r\n', columnSeparator = ';' }: { lineSeparator?: string; columnSeparator?: string }
+    ): boolean {
+        const all_lines = []; // Array of arrays of entries
+        const used_columns = []; // mapped properties to be included
+
+        // initial array of usable text separators. The first character not used
+        // in any text data or as column separator will be used as text separator
+        let ts_list = ['"', "'", '`', '/', '\\', ';', '.', ','];
+
+        if (lineSeparator === columnSeparator) {
+            throw new Error('TODO: lineseparator and columnseparator must differ from each other');
+        }
+
+        ts_list = this.checkCsvTextSafety(lineSeparator, ts_list);
+        ts_list = this.checkCsvTextSafety(columnSeparator, ts_list);
+
+        // create header data
+        const header = [];
+        for (const column of columns) {
+            if (column.title) {
+                ts_list = this.checkCsvTextSafety(column.title, ts_list);
+                header.push(column.title);
```

Comment:
Should this be translated?

Reply:
I would not translate it during csv export, but rather on the other end (the listView), where a translationService is already supposed to be in place

Reply:
You can just put the `TranslateService` into the constructor. Then we have one central place where the values get translated and the views do not have to care about this (should be less code there then).